drivers/sensor/lis2dh/lis2dh.c

```diff
@@ -261,13 +300,105 @@ static const struct sensor_driver_api lis2dh_driver_api = {
+	const struct lis2dh_config *cfg = dev->config;
+	int status;
+	uint8_t raw[6];
+	/* Initialize control register ctrl1 to ctrl 6 to default boot values
+	 * to avoid warm start/reset issues as the accelerometer has no reset
+	 * pin. Register values are retained if power is not removed.
+	 * Default values see LIS2DH documentation page 30, chapter 6.
+	 */
+	(void)memset(raw, 0, sizeof(raw));
+	raw[0] = LIS2DH_ACCEL_EN_BITS;
+
+	status = lis2dh->hw_tf->write_data(dev, LIS2DH_REG_CTRL1, raw,
+						sizeof(raw));
+
+	if (status < 0) {
+		LOG_ERR("Failed to reset ctrl registers.");
+		return status;
+	}
+
+	/* set full scale range and store it for later conversion */
+	lis2dh->scale = lis2dh_reg_val_to_scale[LIS2DH_FS_IDX];
+	status = lis2dh->hw_tf->write_reg(dev, LIS2DH_REG_CTRL4,
+					LIS2DH_FS_BITS | LIS2DH_HR_BIT);
+	if (status < 0) {
+		LOG_ERR("Failed to set full scale ctrl register.");
+		return status;
+	}
+
+	/* set High-pass filter mode */
+	status = lis2dh->hw_tf->write_reg(dev, LIS2DH_REG_CTRL2,
+					(uint8_t)LIS2DH_HPM_BITS);
+	if (status < 0) {
+		LOG_ERR("Failed to set full scale ctrl register.");
+		return status;
+	}
+
+#if CONFIG_LIS2DH_WAKE_THS > 0 && CONFIG_LIS2DH_WAKE_DUR > 0
+	#if CONFIG_LIS2DH_WAKE_THS > LIS2DH_WAKE_THS_ABS_MAX
+		LOG_ERR("Wake Threshold exceeds maximum value of %i",
+		LIS2DH_WAKE_THS_ABS_MAX);
+		return -EINVAL;
+	#elif (CONFIG_LIS2DH_WAKE_THS > LIS2DH_WAKE_THS_RECOMM_MAX)
+		LOG_WRN("Wake Threshold exceeds recommended value of %i at ODR %i \
+threshold can probably never be reached", LIS2DH_WAKE_THS_RECOMM_MAX, LIS2DH_ODR_IDX);
+	#endif
+	
+	status = lis2dh->hw_tf->write_reg(dev, LIS2DH_REG_ACT_TH,
+					(uint8_t)CONFIG_LIS2DH_WAKE_THS);
+
+	if (unlikely(status < 0)) {
+		LOG_ERR("Failed to set Inertial Wake-Up threshold register.");
+		return status;
+	}
+
+	status = lis2dh->hw_tf->write_reg(dev, LIS2DH_REG_ACT_DUR,
+					(uint8_t)CONFIG_LIS2DH_WAKE_DUR);
+
+	if (unlikely(status < 0)) {
+		LOG_ERR("Failed to set Inertial Wake-Up duration register.");
+		return status;
+	}
+#elif CONFIG_LIS2DH_WAKE_THS < 0 || CONFIG_LIS2DH_WAKE_DUR < 0
+	LOG_ERR("CONFIG_LIS2DH_WAKE_THS or CONFIG_LIS2DH_WAKE_DUR was set \
+to a negative value. Both values must be positive!");
+	return -EINVAL;
+#elif (CONFIG_LIS2DH_WAKE_THS == 0 && CONFIG_LIS2DH_WAKE_DUR > 0) || \
+	(CONFIG_LIS2DH_WAKE_THS > 0 && CONFIG_LIS2DH_WAKE_DUR == 0)
+	LOG_ERR("CONFIG_LIS2DH_WAKE_THS or CONFIG_LIS2DH_WAKE_DUR was set \
+			without the other. Both values need to be set!");
+	return -EINVAL;
+#endif
+
+#ifdef CONFIG_LIS2DH_TRIGGER
+	status = lis2dh_init_interrupt(dev);
+	if (unlikely(status < 0)) {
+		LOG_ERR("Failed to initialize interrupts.");
+		return status;
+	}
+#endif
+
+	LOG_INF("bus=%s fs=%d, odr=0x%x lp_en=0x%x scale=%d",
+		    cfg->bus_name, 1 << (LIS2DH_FS_IDX + 1),
+		    LIS2DH_ODR_IDX, (uint8_t)LIS2DH_LP_EN_BIT, lis2dh->scale);
+
+	/* enable accel measurements and set power mode and data rate */
```

Comment:
So are by default all axis used?

Reply:
Please state this clearly in a comment and add that as a todo for the driver (somewhere where you're creating an overview of the functionalities and limitations of the driver's current state, like at the beginning of the driver or the README etc)